src/index.ts

```diff
@@ -164,10 +163,36 @@ class ServerlessCustomDomain {
         credentials.region = this.serverless.providers.aws.getRegion();
 
         this.apiGatewayWrapper = new APIGatewayWrapper(credentials);
-        this.route53 = new this.serverless.providers.aws.sdk.Route53(credentials);
         this.cloudFormationWrapper = new CloudFormationWrapper(credentials);
     }
 
+    /**
+     * Setup route53 resource
+     */
+    public createRoute53Resource(domain: DomainConfig): any {
+      let route53: any;
+
+      const domainProfile = domain.route53Profile;
+      if (domainProfile) {
+        const route53Credentials = new this.serverless.providers.aws.sdk.SharedIniFileCredentials({
+          profile: domainProfile,
+        });
+        const route53Region = domain.route53Region || this.serverless.providers.aws.getRegion();
+
+        route53 = new this.serverless.providers.aws.sdk.Route53({
+          credentials: route53Credentials,
+          region: route53Region,
```

Comment:
Is this needed since Route53 is a global service?

Reply:
Good question. I'm actually not 100% sure. 

Generally I would say it isn't needed, since most regions point to one and the same endpoint (route53.amazonaws.com). There are however two exceptions to this; the Chinese regions. Those regions point to another endpoint. I think if you need to manage your domains in one of these regions, but you deploy from outside of these regions, you might need to actually set the region for Route53. However, as said, I'm not 100% sure on this. Since the region is optional and will assume the region specified with serverless if not otherwise set, I reckoned this was the safest approach. I'd be happy to change things around however if my assumption is not correct and/or you think otherwise.

Let me know...